internal/export/distro/format.go

```diff
@@ -275,3 +280,63 @@ func (af biotFormatter) Format(event *models.Event) []byte {
 	}
 	return msg
 }
+
+// senMLJSONFormatter is used to convert Event to SenML message and
+// SenML message to bytes.
+type senMLJSONFormatter struct {
+}
+
+// SenML JSON formatter
+func (senMLjsonTr senMLJSONFormatter) Format(event *models.Event) []byte {
+
+	readings := []senml.SenMLRecord{}
+
+	for _,v := range event.Readings {
+
+		reading := senml.SenMLRecord{}
+
+		reading.BaseName = v.Device + ".edgex-gateway"
+		reading.Name = v.Name
+		reading.Unit = v.Unit
+		reading.Time = float64(v.Origin/1000)
+
+
+		switch  {
+		case v.Type == models.String:
+			reading.StringValue = v.Value
+		case v.Type == models.Bool:
+			b,_ := strconv.ParseBool(v.Value)
+			reading.BoolValue = &b
+		case v.Type == models.Float64:
+			b, err := base64.StdEncoding.DecodeString(v.Value)
+			if err != nil {
+				LoggingClient.Error(fmt.Sprintf("Error parsing SenML Float value. Error: %s", err.Error()))
+			} else {
+				value := Float64frombytes(b)
+				reading.Value = &value
+			}
+			// SenML Data not supported yet
+		}
```

Comment:
Do you want a default case in here if the `v.Type` isn't one of the above specific types? Should that default populate v.DataValue?

Also @SteveOss @tonyespy notice the assumption that `float` is encoded to `base64` above. This PR is now pending a decision on that representation.